src/Forms/ConfirmedPasswordField.php

```diff
@@ -753,4 +847,36 @@ private function getAttributesHTMLForChild(FormField $child): DBField
         }
         return DBField::create_field('HTMLFragment', $attributes);
     }
+
+    /**
+     * Get the applicable password validator based on if we are on a member form or not
+     */
+    private function getApplicableValidator(): ?PasswordValidator
+    {
+        if ($this->getIsOnMemberForm()) {
+            return Member::password_validator();
+        }
+        $spec = Injector::inst()->getServiceSpec(PasswordValidator::class);
+        // Do not attempt to instantiate abstract class PasswordValidator
+        // This can happen during unit tests when the member password validator is set to null
+        if ($spec && $spec['class'] !== PasswordValidator::class) {
+            return PasswordValidator::singleton();
+        }
+        return null;
+    }
+
+    /**
+     * Whether password strength evaulation is possible in the current context
+     */
+    private function canEvaluateStrength(): bool
+    {
+        if (!$this->getIsOnMemberForm() && !$this->getRequireStrongPassword()) {
+            return false;
+        }
+        $validator = $this->getApplicableValidator();
+        if ($validator) {
+            return $validator->canEvaluateStrength();
+        }
+        return false;
```

Comment:
```suggestion
    if ($this->getRequireStrongPassword()) {
        return true;
    }
    return $this->getIsOnMemberForm() && is_a(Member::password_validator(), EntropyPasswordValidator::class);
```
We've agreed that only `EntropyPasswordValidator` is relevant for member forms, so we can evaluate strength if:
1. We require a strong password on this field explicitly (regardless of whether this is on a member form or not), or
2. we're on a member form and the member is using an `EntropyPasswordValidator`.

See also https://github.com/silverstripe/silverstripe-framework/pull/11417#discussion_r2160516481 which suggests an alternative approach to this method

Reply:
> We've agreed that only EntropyPasswordValidator is relevant for member forms,

I'm not sure what you're referring to here

`Member::set_password_validator(?PasswordValidator $validator = null)` allows setting a non-EntropyPasswordValidator so we need too account for that?

Reply:
I've removed the whole method